### PR TITLE
enable 'editData' and 'createViewFromContainer' for Container

### PR DESCRIPTION
**Pull request by Yassir-BenBOUBKER**

### CODE REVIEW

File: src/main/webapp/js/diagramly/Menus.js [hunk 1]
```diff
@@ -3725,7 +3725,7 @@
 		this.put('edit', new Menu(mxUtils.bind(this, function(menu, parent)
 		{
 			this.addMenuItems(menu, ['undo', 'redo', '-', 'cut', 'copy', 'copyAsImage', 'paste',
-				'delete', '-', 'duplicate', '-', 'findReplace', '-', 'editData', 'editTooltip', '-',
+				'delete', '-', 'duplicate', '-', 'findReplace', '-', 'editData','createViewFromContainer', 'editTooltip', '-',
```
Comment (Thebarda) [inline src/main/webapp/js/diagramly/Menus.js]:
```suggestion
				'delete', '-', 'duplicate', '-', 'findReplace', '-', 'editData', 'createViewFromContainer', 'editTooltip', '-',
```

File: src/main/webapp/js/grapheditor/Actions.js [hunk 2]
```diff
@@ -725,13 +727,47 @@ Actions.prototype.init = function()
 		if (cell.getAttribute('widgetType') !== undefined) {
 		 	cellAttributes = createAttributes(propsWidget);
 		}
+		if (cell.getAttribute('type') === 'CONTAINER') {
+			cellAttributes = createAttributes(propsContainer);
+		}
+
+		console.log('Action');
```
Comment (Thebarda) [inline src/main/webapp/js/grapheditor/Actions.js]: Remove console logs

File: src/main/webapp/js/grapheditor/Actions.js [hunk 3]
```diff
@@ -725,13 +727,47 @@ Actions.prototype.init = function()
...
+		return arrayProps.map((key) => {
+			if (cell.getAttribute(key) !== undefined) {
+				return {[key]: cell.getAttribute(key)};
+			}
+		}).filter(prop => prop !== undefined);
+	};
+	
+	if (cell.getAttribute('type') === 'CONTAINER') {
+
+		console.log('CreateViewFromContainer')
```
Comment (jdelpierre) [inline src/main/webapp/js/grapheditor/Actions.js]: remove console.log()
```suggestion
```

File: src/main/webapp/js/grapheditor/Actions.js [hunk 4]
```diff
@@ -725,13 +727,47 @@ Actions.prototype.init = function()
 		if (cell.getAttribute('widgetType') !== undefined) {
 		 	cellAttributes = createAttributes(propsWidget);
 		}
+		if (cell.getAttribute('type') === 'CONTAINER') {
+			cellAttributes = createAttributes(propsContainer);
+		}
+
+		console.log('Action');
+		console.log(cellAttributes);
+		console.log('Action');
```
Comment (jdelpierre) [inline src/main/webapp/js/grapheditor/Actions.js]: remove console.log()
```suggestion
```

File: src/main/webapp/js/grapheditor/Actions.js [hunk 5]
```diff
@@ -699,6 +700,7 @@ Actions.prototype.init = function()
 			'metricsColorStart', 'metricsColorEnd', 'parentName', 'parentType', 'parentId',
 			'resourceName', 'orientation',
 		];
+		const propsContainer = ['type', 'modelId', 'viewId', 'containerName', 'centreonImageId'];
```
Comment (Thebarda) [inline src/main/webapp/js/grapheditor/Actions.js]:
```suggestion
		const containerProps = ['type', 'modelId', 'viewId', 'containerName', 'centreonImageId'];
```

File: src/main/webapp/js/grapheditor/Actions.js [hunk 6]
```diff
@@ -725,13 +727,40 @@ Actions.prototype.init = function()
 		if (cell.getAttribute('widgetType') !== undefined) {
 		 	cellAttributes = createAttributes(propsWidget);
 		}
+		if (cell.getAttribute('type') === 'CONTAINER') {
+			cellAttributes = createAttributes(propsContainer);
```
Comment (Thebarda) [inline src/main/webapp/js/grapheditor/Actions.js]:
```suggestion
			cellAttributes = createAttributes(containerProps);
```

File: src/main/webapp/js/grapheditor/Actions.js [hunk 7]
```diff
@@ -725,13 +727,40 @@ Actions.prototype.init = function()
...
+	function createAttributes(arrayProps) {
+		return arrayProps.map((key) => {
+			if (cell.getAttribute(key) !== undefined) {
+				return {[key]: cell.getAttribute(key)};
+			}
+		}).filter(prop => prop !== undefined);
+	};
+	
+	if (cell.getAttribute('type') === 'CONTAINER') {
+		cellAttributes = createAttributes(propsContainer);
```
Comment (Thebarda) [inline src/main/webapp/js/grapheditor/Actions.js]:
```suggestion
		cellAttributes = createAttributes(containerProps);
```

**Review by Thebarda — APPROVED** [review]

**Review by jdelpierre — APPROVED** [review]

**Pull request merged**